eng/scripts/Invoke-GenerateAndBuildV2.ps1

```diff
@@ -108,6 +109,39 @@ if ($inputFileToGen) {
     UpdateExistingSDKByInputFiles -inputFilePaths $inputFileToGen -sdkRootPath $sdkPath -headSha $commitid -repoHttpsUrl $repoHttpsUrl -downloadUrlPrefix "$downloadUrlPrefix" -generatedSDKPackages $generatedSDKPackages
 }
 
+# generate sdk from cadl file
+if ($relatedCadlProjectFolder) {
+    for ($i = 0; $i -le $relatedCadlProjectFolder.Count - 1; $i++) {
+        $caldFolder = (Join-Path $swaggerDir $relatedCadlProjectFolder[$i]) -replace "\\", "/"
+        $newPackageOutput = "newPackageOutput.json"
+
+        Push-Location $caldFolder
+        $cadlProjectYaml = Get-Content -Path "$caldFolder/cadl-project.yaml" -Raw
+
+        Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+        $yml = ConvertFrom-YAML $cadlProjectYaml
+        $sdkFolder = $yml["emitters"]["@azure-tools/cadl-csharp"]["sdk-folder"]
+        $projectFolder = (Join-Path $sdkPath $sdkFolder)
+        $projectFolder = $projectFolder -replace "\\", "/"
+        if ($projectFolder) {
+            $directories = $projectFolder.Split("/");
+            $count = $directories.Count
+            $projectFolder = $directories[0 .. ($count-2)] -join "/"
+            $service = $directories[-3];
+            $namespace = $directories[-2];
+        }
+        New-CADLPackageFolder -service $service -namespace $namespace -sdkPath $sdkPath -cadlInput $caldFolder/main.cadl -outputJsonFile $newpackageoutput
+        $newPackageOutputJson = Get-Content $newPackageOutput | Out-String | ConvertFrom-Json
+        $relativeSdkPath = $newPackageOutputJson.path
+        npm install
```

Comment:
Is there a reason you run `npm install` before `npx`? If it's just for cadl tools, it's unnecessary. `npx` will fetch modules as needed and run the specified package's entry point. If it's only for cadl, I imagine `npm install` will install much more than needed as well.

Reply:
`npx cadl compile` only to fetch cadl compile (@cadl-liang/compiler) modules, but each cadl project will depend on a group of modules and may different for each cadl project. There is an package.json in the cadl project, we need to `npm install` to install all the depend modules this cadl project needed.